conans/client/package_layouts/package_base_layout.py

```diff
@@ -0,0 +1,19 @@
+# coding=utf-8
+
+import os
+
+from conans.model.ref import ConanFileReference
+
+
+class PackageBaseLayout(object):
+    """ User customizable layout for packages """
+
+    def __init__(self, base_folder, conan_ref, short_paths=False):
```

Comment:
short_paths might be a mechanism of the cache only.

Reply:
I understand you, but I think we have to change our mind around it. If we want to delegate the path retrieval to the `Package*Layout` objects we need to allow them to know if short_paths should be used or not. Only the `PackageCacheLayout` will use `short_paths` and the `ClientCache` itself should not know which package layout is being used (we are accessing through the cache both layouts, the _original cache_ one and the _user workspace_ one).

Anyway, there is a long way to go. I need to explore the POC with the editable packages in order to get a better idea of the final layout of classes, methods, files,...

Reply:
`short_paths` is a mechanism of the `PackageCacheLayout` only